timemachine/maps/terminal_bonds.py

```diff
@@ -0,0 +1,239 @@
+"""Summarize HarmonicBonds as Intervals in R+, construct maps between Intervals, & apply these maps to bond lengths"""
+
+from dataclasses import dataclass
+from typing import Tuple
+
+import numpy as np
+from jax import config, jacobian, jit
+from jax import numpy as jnp
+from jax import vmap
+from numpy.typing import NDArray as Array
+
+config.update("jax_enable_x64", True)
+
+from typing import List
+
+import networkx as nx
+
+from timemachine.constants import BOLTZ, DEFAULT_TEMP
+
+
+@dataclass
+class Interval:
+    lower: float
+    upper: float
+
+    @property
+    def width(self) -> float:
+        return self.upper - self.lower
+
+    def validate(self) -> None:
```

Comment:
Validation seems better done at construction (unless there's a good reason to defer?). For dataclasses, can do validation in the `__post_init__` method ([example](https://stackoverflow.com/a/60179826))

Reply:
Thank you -- your suggestion is clearly better!

>unless there's a good reason to defer?

No reason -- just ignorance of the `__post_init__` method

Reply:
Addressed in https://github.com/proteneer/timemachine/pull/801/commits/28379e03a404d1f032b40b22fcb2175c336f6451